source/adapters/level_zero/usm.cpp

```diff
@@ -187,8 +187,15 @@ static ur_result_t USMDeviceAllocImpl(void **ResultPtr,
     ZeDesc.pNext = &RelaxedDesc;
   }
 
-  ZE2UR_CALL(zeMemAllocDevice, (Context->ZeContext, &ZeDesc, Size, Alignment,
```

Comment:
Looks like this macro was needed for the l0 api tracing enabled with `UR_L0_DEBUG`, that tracing it necessary for a lot of the sycl-e2e tests (they check for certain l0 api calls in the trace output). We should make these kinds of changes like this
```
ze_result_t ZeResult = ZE_CALL_NOCHECK(zeMemAllocDevice, (Context->ZeContext,
              &ZeDesc, Size, Alignment, Device->ZeDevice, ResultPtr));
```
instead